jsonnet/components/alertmanager-user-workload.libsonnet

```diff
@@ -0,0 +1,341 @@
+        ],
+      },
+    },
+
+
+    // In order for kube-rbac-proxy to perform a TokenReview and
+    // SubjectAccessReview for authN and authZ the alertmanager ServiceAccount
+    // requires the `create` action on both of these.
+
+    clusterRole: {
+      apiVersion: 'rbac.authorization.k8s.io/v1',
+      kind: 'ClusterRole',
+      metadata: {
+        name: 'alertmanager-' + cfg.name,
+      },
+      rules: [
+        {
+          apiGroups: ['authentication.k8s.io'],
+          resources: ['tokenreviews'],
+          verbs: ['create'],
+        },
+        {
+          apiGroups: ['authorization.k8s.io'],
+          resources: ['subjectaccessreviews'],
+          verbs: ['create'],
+        },
+        {
+          // By default authenticated service accounts are assigned to the `restricted` SCC which implies MustRunAsRange.
+          // This is problematic with statefulsets as UIDs (and file permissions) can change if SCCs are elevated.
+          // Instead, this sets the `nonroot` SCC in conjunction with a static fsGroup and runAsUser security context below
+          // to be immune against UID changes.
+          apiGroups: ['security.openshift.io'],
+          resources: ['securitycontextconstraints'],
+          resourceNames: ['nonroot'],
+          verbs: ['use'],
+        },
+      ],
+    },
+
+    clusterRoleBinding: {
+      apiVersion: 'rbac.authorization.k8s.io/v1',
+      kind: 'ClusterRoleBinding',
+      metadata: {
+        name: 'alertmanager-' + cfg.name,
+      },
+      roleRef: {
+        apiGroup: 'rbac.authorization.k8s.io',
+        kind: 'ClusterRole',
+        name: 'alertmanager-' + cfg.name,
+      },
+      subjects: [{
+        kind: 'ServiceAccount',
+        name: 'alertmanager-' + cfg.name,
+        namespace: cfg.namespace,
+      }],
+    },
+
+    kubeRbacProxyTenancySecret: {
+      apiVersion: 'v1',
+      kind: 'Secret',
+      metadata: {
+        name: 'alertmanager-kube-rbac-proxy-tenancy',
+        namespace: cfg.namespace,
+        labels: { 'app.kubernetes.io/name': 'alertmanager-' + cfg.name },
+      },
+      type: 'Opaque',
+      stringData: {
+        'config.yaml': std.manifestYamlDoc({
+          authorization: {
+            rewrites: {
+              byQueryParameter: {
+                name: 'namespace',
+              },
+            },
+            resourceAttributes: {
+              apiGroup: 'monitoring.coreos.com',
+              resource: 'prometheusrules',
+              namespace: '{{ .Value }}',
+            },
+          },
+        }),
+      },
+    },
+
+    kubeRbacProxyMetricSecret: generateSecret.staticAuthSecret(cfg.namespace, cfg.commonLabels, 'alertmanager-kube-rbac-proxy-metric') + {
+      metadata+: {
+        labels: { 'app.kubernetes.io/name': 'alertmanager-' + cfg.name },
+      },
+    },
+
+    // These patches inject the oauth proxy as a sidecar and configures it with
+    // TLS.
+    alertmanager+: {
+      spec+: {
+        securityContext: {
```

Comment:
This is also not complying with the the new PodSecurity feature, the UWM namespace is under the `privileged` profile. We need to:

1. Add the `seccompProfile` here
2. For each container in the pod add `allowPrivilegeEscalation` and `capabilities`